src/widgets/scrollmessagebox.cpp

```diff
@@ -0,0 +1,143 @@
+/* This file is part of Clementine.
+   Copyright 2010, David Sansome <me@davidsansome.com>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "scrollmessagebox.h"
+#include <QStyle>
+#include <QApplication>
+#include <QPushButton>
+#include <QGridLayout>
+#include <QtGui/QScrollArea>
+
+ScrollMessageBox::ScrollMessageBox(QMessageBox::Icon icon, QString const& title, 
+  QString const& text, QDialogButtonBox::StandardButtons buttons /*= QDialogButtonBox::Ok*/, 
+  QWidget* parent /*= 0*/) :
+  QDialog(parent, Qt::Dialog | Qt::WindowTitleHint | Qt::CustomizeWindowHint | Qt::WindowCloseButtonHint) {
+
+  QLabel *iconLabel;
```

Comment:
Please follow the style guide: https://google.github.io/styleguide/cppguide.html

e.g. `QLabel* icon_label`